### DISCUSSION TITLE
should-you-unit-test-controllers-in-aspnetcore/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#340] [Announcements]
# should-you-unit-test-controllers-in-aspnetcore/

In this post I discuss unit testing of API/MVC controllers in ASP.NET Core and some of the difficulties you face.

https://andrewlock.net/should-you-unit-test-controllers-in-aspnetcore/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at August 29 2020, 06:02, in reply to [Jeff Bowman](https://disqus.com/by/disqus_uTBpSkYgMI/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/340#discussioncomment-6059493)</em>

---

<p>That's a typo, thanks. Fixed it now :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 15 2020, 04:14, in reply to [David V. Corbin](https://disqus.com/by/davidvcorbin/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/340#discussioncomment-6059499)</em>

---

<blockquote>Unfortunately the current state of .Net Core (aka .Net 5.0) is such that many things which were possible to isolate boundaries in the Full Framework are no longer feasible</blockquote><p></p><p>Do you have an example? In my experience the opposite is true - it's much easier to test ASP.NET Core than ASP.NET (on full framework)</p>

**Comment by noibi** [comment]
I'm with you :)
Nice Post!
